OMEdit/OMEditLIB/Modeling/Model.cpp

```diff
@@ -1128,6 +1179,92 @@ namespace ModelInstance
     return false;
   }
 
+  bool Model::isValidConnection(const Name &lhsConnector, const Name &rhsConnector) const
+  {
+    const Element *lhs = lookupElement(lhsConnector);
+    const Element *rhs = lookupElement(rhsConnector);
+
+    if (!lhs || !rhs) {
+      qDebug() << "Failed to find connector " << (lhs ? rhsConnector : lhsConnector).getName();
+      return true;
+    }
+
+    // An input should not be connected to an input, or an output to an output.
+    auto dir = lhs->getDirection();
+    if (!dir.isEmpty() && dir == rhs->getDirection()) {
+      return false;
+    }
+
+    // Check that the connectors are type compatible.
+    Model *lhs_model = lhs->getModel();
+    Model *rhs_model = rhs->getModel();
+
+    if (!lhs_model || !rhs_model) return false;
+
+    return lhs_model->isTypeCompatibleWith(*rhs_model);
+  }
+
+  bool Model::isTypeCompatibleWith(const Model &other) const
+  {
+    if (isExpandableConnector() || other.isExpandableConnector()) {
+      // Don't type check expandable connectors, since we don't really know what
+      // they contain here.
+      return true;
+    } else if (isConnector() && other.isConnector()) {
```

Comment:
We need to check for basic types of connectors as well. Otherwise it will allow connections like `RealInput -> BooleanOutput`

Reply:
That's handled in `Model::isValidConnection`, since we need the `Element` for the connector rather than the `Model` to be able to check that.

Reply:
Are you sure? I just tested and I can connect RealInput to BooleanOutput.

Reply:
How did you check? I tested with e.g. two `Modelica.Blocks.Logical.Greater` and connecting the outputs together, and that gives an error as expected.

Reply:
I connected `Modelica.Blocks.Interfaces.RealInput` with `Modelica.Blocks.Interfaces.BooleanOutput`

Reply:
Shall we also change it so it doesn't allow making a connection. Right now it only prints a message.

Reply:
And we also need to check for connections in `CreateConnectionDialog::createConnection`. This is used when any of the connector is an array.

Reply:
I got a bit confused and thought the issue you had was with the direction of the connectors, but the actual issue was that they were different types. #10723 should fix that issue, now we check that connectors that are derived from types are actually of the same type.

Reply:
#10724 adds a check for array connections too.

As for changing it to not allow invalid connections I'm still not sure, but you can just change it yourself if you think it's a good idea since you know better how OMEdit works.